lib/model.js

```diff
@@ -411,6 +413,61 @@ assign(derived, {
   }
 });
 
+/**
+ * Adds auth info to URL. The connection model builds two URLs.
+ * driverUrl - for the driver with the password included.
+ * safeUrl - for the UI with stars instead of password.
+ *
+ * @param {Object} options - Has only isPasswordProtected propery.
+ *
+ * @returns {String} - URL with auth.
+ */
+function addAuthToUrl({ isPasswordProtected }) {
+  let username = '';
+  let password = '';
+  let authField = '';
+  let result = this.baseUrl;
+
+  // Post url.format() workaround for
+  // https://github.com/nodejs/node/issues/1802
+  if (
+    this.authStrategy === 'MONGODB' ||
+    this.authStrategy === 'SCRAM-SHA-256'
+  ) {
+    username = encodeURIComponent(this.mongodbUsername);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.mongodbPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'LDAP') {
+    username = encodeURIComponent(this.ldapUsername);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.ldapPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'X509') {
+    username = encodeURIComponent(this.x509Username);
+    authField = username;
+  } else if (this.authStrategy === 'KERBEROS' && this.kerberosPassword) {
+    username = encodeURIComponent(this.kerberosPrincipal);
+    password = isPasswordProtected ? '*****' : encodeURIComponent(this.kerberosPassword);
+    authField = format('%s:%s', username, password);
+  } else if (this.authStrategy === 'KERBEROS') {
+    username = encodeURIComponent(this.kerberosPrincipal);
+    authField = format('%s:', username);
+  }
+
+  // The auth component comes straight after `the mongodb://`
+  // so a single string replace should always work
+  result = result.replace('AUTH_TOKEN', authField, 1);
+
+  if (['LDAP', 'KERBEROS'].includes(this.authStrategy)) {
```

Comment:
We might want to use the lodash `includes` here instead of the javascript prototype one, so that it polyfills on IE.
https://github.com/mongodb-js/connection-model/blob/master/lib/model.js#L523 is an example

Reply:
Hm do you think someone can use a connection model outside of Compass. Since Compass is an electron app we don't care about IE much.

Reply:
Yeah I don't think it matters. That being said there might be a case for consistency in the file. lgtm tho :shipit:

Reply:
Consistency is a good point for me :) updated!